<?xml version="1.0"?>
<launch>
  <arg name="pointcloud_topic" doc="ROS topic for pointcloud input"/>
  <arg name="debug_cg" default="false" doc="Run the coxgraph client in debug mode." />
  <arg name="debug_tsdf" default="false" doc="Run the tsdf client in debug mode." />
  <arg name="show_rviz" default="false" doc="Automatically launch Rviz." />
  <arg name="verbose" default="false"/>

  <group ns="coxgraph">
    <arg unless="$(arg debug_cg)" name="launch_prefix" value="" />
    <arg if="$(arg debug_cg)" name="launch_prefix" value="xterm -e gdb --args" />

    <node name="coxgraph_client_0" pkg="coxgraph" type="coxgraph_client_node" output="screen" launch-prefix="$(arg launch_prefix)" args="-alsologtostderr">
      <rosparam file="$(find voxgraph)/config/voxgraph_mapper.yaml" />
      <param name="input_odom_frame" value="map_0"/>
      <param name="input_base_link_frame" value="robot_base_0"/>
      <param name="output_mission_frame" value="mission_0"/>
      <param name="output_odom_frame" value="map_0"/>
      <param name="output_base_link_frame" value="robot_base_0"/>
      <param name="output_sensor_frame" value="sensor_coxgraph_0"/>
      <param name="tsdf_voxel_size" value="0.05"/>
      <param name="tsdf_integrator/truncation_distance" value="0.15"/>
      <param name="submap_topic" value="/tsdf_client/label_tsdf_client_0/tsdf_map_out" />
      <param name="loop_closure_topic" value="/coxgraph/coxgraph_client_0/loop_closure_in" />
      <param name="client_id" value="0"/>
    </node>
  </group>

  <group ns="tsdf_client">
    <include file="$(find coxgraph)/launch/cvg/tsdf_client0_cvg.launch">
      <arg name="debug" default="$(arg debug_tsdf)"/>
      <arg name="verbose" default="$(arg verbose)"/>
    </include>
  </group>

</launch>
